tests/contest/contest/src/utils/test_utils.rs

```diff
@@ -222,7 +222,12 @@ pub fn test_inside_container(
         .wait_with_output()
     {
         Ok(c) => c,
-        Err(e) => return TestResult::Failed(anyhow!("container start failed : {:?}", e)),
+        Err(e) => {
+            // given that start has failed, we can be pretty sure that create has either failed
+            // or completed already, so we wait on it so it does not become a zombie process
+            let _ = create_process.wait_with_output();
```

Comment:
Make sure the `wait_with_output` does not block when the create process is failed.

Reply:
Hey, I'm pretty certain that if create process has failed, the wait will not block. It will probably return an error, but it will not block. As an added safety, I have also changed the code to explicitly kill the create_process before waiting on it ; in the case where create is successful, but start fails for some reason. In such case wait will block forever, so killing it will help it.
Can you take another look at it? Thanks :)